### ISSUE TITLE
Explore replacing react-markmirror with react-mde

### ISSUE THREAD

**Issue by go1dfish** [decred/politeiagui#114]
I think overall this editor is better suited for our purposes, it's lighter and doesn't have weird text duplication bugs on mobile browsers.

It doesn't do fancy syntax highlighting though, 

Need this to happen before the previews will be accurate though:

https://github.com/andrerpena/react-mde/issues/56

**Comment by fernandoabolafio** [comment]
working on it

**Comment by fernandoabolafio** [comment]
From what I've learned about it so far I can say that we can actually use custom markdown parsers with `react-mde`. See that from their source code you can compose it like this:
```
[...]
 return (
            <div className="react-mde">
                <ReactMdeToolbar
                    commands={commands}
                    onCommand={this.handleCommand}
                />
                <ReactMdeTextArea
                    onChange={this.handleValueChange}
                    value={value}
                    textAreaProps={textAreaProps}
                    textAreaRef={(c) => this.textArea = c}
                />
                <ReactMdePreview
                    markdown={value ? value.text : ""}
                    previewRef={(c) => this.preview = c}
                    showdownOptions={showdownOptions}
                />
            </div>
)
```
So its totally tangible to replace the `ReactMdePreview` by another custom markdown of our own preference. However, their markdown parser is pretty good. I've tested and it support all features listed on `proposals/new`: 
<img width="526" alt="screen shot 2017-12-04 at 12 48 36 pm" src="https://user-images.githubusercontent.com/14864439/33558641-90b31bd8-d8f1-11e7-923c-678bdc2a157f.png">

An issue we will face migrating to `react-mde` is that their 'native styles' are conflicting with the styles on `politeiagui/src/style/theme/index.css` because of globally scoped styles such as `body button {...} `. So I need to customize the markdown editor styles to not be overridden by those. 

In summary: we can fork `react-mde` to solve styles conflicts, enable better (and easy) customization for toolbar buttons and their functionalities and grow from there. From what I've seen is a good change since it works better on mobile and is lighter than markmirrow.

**Comment by sndurkin** [comment]
@fernandoabolafio, in the future, please put the "Fixes <issue>" text in the pull request description so that when the PR is merged the issues are automatically closed. It doesn't seem to work if you add the text in a comment, as you did here: https://github.com/decred/politeiagui/pull/128#issuecomment-349152247
